runtime/vm/lookupmethod.c

```diff
@@ -1012,8 +1012,13 @@ getModuleNameUTF(J9VMThread *currentThread, j9object_t	moduleObject, char *buffe
 		nameBuffer = buffer;
 #undef UNNAMED_MODULE
 	} else {
-		nameBuffer = copyStringToUTF8WithMemAlloc(
-			currentThread, module->moduleName, J9_STR_NULL_TERMINATE_RESULT, "", 0, buffer, bufferLength, NULL);
+		PORT_ACCESS_FROM_VMC(currentThread);
+		J9UTF8 *moduleName = module->moduleName;
+		U_8 *nameData = J9UTF8_DATA(moduleName);
+		UDATA nameLength = J9UTF8_LENGTH(moduleName);
+		nameBuffer = (char *)j9mem_allocate_memory(nameLength + 1, OMRMEM_CATEGORY_VM);
```

Comment:
`malloc`'s failure case needs to be addressed.

Reply:
Missed this. Thanks.

Reply:
Maybe we can introduce copyJ9UTF8ToUTF8WithMemAlloc, similar to copyStringToUTF8WithMemAlloc, which could be used in most places?

Reply:
The problem is that when we add the rest of the support for snapshotting modules, we need to allcocate the `moduleName` `J9UTF8`s with the snapshot sub-allocator.

The API could just take a function pointer to the allocator to get around that, or we could have a snapshot specific function for when snapshots are enabled.

Reply:
> The API could just take a function pointer to the allocator to get around that

This one seems better (less code). Instead of a function pointer, we can also use a boolean to toggle between the allocators.

Reply:
> we can also use a boolean to toggle between the allocators

To keep it general/reusable beyond snapshotting, I think it is better to have the allocator as a parameter rather than a boolean to switch between the two.

But here, I will use the functions that we already have, and add a `copyJ9UTF8ToUTF8WithMemAlloc` function as if we were ignorant of the future module snapshotting support. Then in the PR that adds modules to the snapshots, I will add a variant that takes an allocator pointer and use that anywhere we need.